docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+
+```
+OpenTimelineIO/
+    `- src/
+        `- opentime/
+        `- opentimelineio/
+        `- openedit/
+            `- ...
+```
+
+The benefit for the extra library is the edit suite will inevitably grow and will begin crowding the core model, which ultimately should be able to run without it.
+
+## Dependencies / Namespaces
+The `openedit` library will have to communicate with both `opentime` and `openedit`.
+
+This will warrant either a new namespace `namespace openedit { ... }` or using `opentimelineio` again. The latter makes some of the bellow comments less challenging but again, might pollute an otherwise clean namespace.
+
+### ErrorStatus
+This will present challenges when interfacing with the `ErrorStatus` of the different namespaces. That said, with a simple wrapper for this, we could have the instances auto convert as required.
+
+### Retainer
+Ideally, we continue to use the `Retainer<>` where possible to make sure editing is reference counted. Namespaces might make the names a bit clunky, seeing `SerializedObject::Retainer<>` all over:
+
+```
+namespace openedit { OPENEDIT_VERSION {
+
+using namespace opentimelineio::OPENTIMELINEIO_VERSION;
+auto item = SerializedObject::Retainer<Item>(...);
+
+} }
+```
+
+So perhaps establishing some common aliases early might help?
+```
+using RetainedItem = SerializedObject::Retainer<Item>;
+using RetainedComposition = SerializedObject::Retainer<Composition>;
+...
+```
+
+# Editing Commands
+Let's go through some of the most common commands and what the python API might look like:
+
+## Overwrite
+
+![Overwrite](../_static/edit/01_overwrite.png)
+- Anything overlapping is destroyed on contact. This may split an item.
+
+#### API
+```py
+"""
+Args:
+    item: Item that we're going to place onto the track
+    track: Track that this item will belong to afterwards
+    track_time: RationalTime of our track to put this item at
+    fill_template: Optional[Item] that will be cloned where required
+        to fill in the event that this overwrite extends the end of the
+        composition's limit
+"""
+otio.algorithms.overwrite(
+    item: Item,
+    track: Track,
+    track_time: RationalTime,
+    fill_template: Item = None, # Default to Gap
+)
+```
+- In the image, Clip `C` is `item` and `track` is the what `A` and `B` are on.
+
+----
+
+## Insert
+![Insert](../_static/edit/02_insert.png)
+- Items past the in point are shifted by the new items duration.
+- Item may be split if required.
+
+#### API
+```py
+"""
+Args:
+    <same as overwrite>
+"""
+otio.algorithms.insert(
+    item: Item,
+    track: Track,
+    composition_time: RationalTime,
+    fill_template: Item = None, # Default to Gap
+)
+```
+
+---
+
+## Trim
+![Trim](../_static/edit/03_trim.png)
+- Adjust a single item's start time or duration.
+- Do not affect other clips.
+- Fill now-empty time with gap or template.
+- Clamps source_range to non-gap boundary (can only overwrite gaps)
+
```

Comment:
if source range isn't set, does the available range become the source range before modifying the values?

This is assuming the item is in track, is it an error condition if the item has no parent?

If the item is directly in a stack, is a track inserted wrapping the item so that gap can be put before or after the item?

Reply:
I hope that the common commands are as simple as possible - with explicitly defined implementation or course. Part of that would be, possibly, having it assume the source_range is the available_range if not set before. Implementing it as an error would totally work too but do we gain anything with that?